src/core/service.c

```diff
@@ -4908,6 +4911,24 @@ static void service_handoff_timestamp(
         unit_add_to_dbus_queue(u);
 }
 
+static void service_notify_pidref(Unit *u, PidRef *parent_pidref, PidRef *child_pidref) {
+        Service *s = ASSERT_PTR(SERVICE(u));
+
+        assert(parent_pidref);
+        assert(child_pidref);
+
+        if (pidref_equal(&s->main_pid, parent_pidref)) {
+                pidref_done(&s->main_pid);
+                s->main_pid = TAKE_PIDREF(*child_pidref);
+        } else if (pidref_equal(&s->control_pid, parent_pidref)) {
+                pidref_done(&s->control_pid);
+                s->control_pid = TAKE_PIDREF(*child_pidref);
```

Comment:
This is not watched through unit_watch_pidref?

Reply:
Yea seems we should, this is missing